### PR TITLE
New OSE3 docker host builder and OpenStack ansible provisioning support

### PR DESCRIPTION
**Pull request by sabre1041**
#### What does this PR do?

Adds docker container with `atomic-openshift-utils` to enable provisioning of OpenShift environments plus initial work on Ansible tooling (Support for provisioning new OpenStack instances that will in the future run OpenShift) 
#### How should this be manually tested?

Since this PR contains two features, the following instructions can be used to test and validate

Pull down the contents of the PR
### Docker Container

On a RHEL7 machine with appropriate entitlements for RHEL7 and OSE3, execute the following command in the `docker/openshift-host-builder` folder to build the container:

```
./run.sh --repository=<path_to_rhc-ose_repo> --ssh=<directory_containing_ssh_keys_for_openstack>
```

Once the container completes building and you are inside a new container, validate the following file is present to confirm the successful installation of the `atomic-openshift-utils` package

```
/usr/share/ansible/openshift-ansible/playbooks/byo/config.yml
```
### Ansible Automation

This PR contains the initial work on migrating the provisioning scripts to Ansible. The primary goals addressed are provisioning 3 instances in OpenStack (representing 1 master and 2 nodes), create and attach a storage volume and to update the machines with the most up to date packages. Additional work will be forthcoming on the next set of tasks.

To validate, set the `openstack_key_name` property in the `ose3` inventory file to the name of your OpenStack security key.

Run the following command from the `rhc-ose-ansible` folder

```
ansible-playbook -i ose3 site.yml
```

Validate 3 machines were successfully created in OpenStack with a new volume attached at /dev/vdb 
#### Is there a relevant Issue open for this?

No.
#### Who would you like to review this?

/cc @etsauer @oybed @JayKayy

### CODE REVIEW

File: rhc-ose-ansible/roles/openstack-create/defaults/main.yml [hunk 1]
```diff
@@ -0,0 +1,9 @@
+---
+env_id: "testenv"
+openstack_master_security_group: "ose3-master"
+openstack_node_security_group: "ose3-node"
+openstack_flavor_id: "d5fbd66f-2069-42d6-9bde-254bac4617b1"
```
Comment (oybed) [inline rhc-ose-ansible/roles/openstack-create/defaults/main.yml]: Any chance we can use the flavor name rather than the id here?

**Comment by oybed** [comment]
@sabre1041 Looks great for a start. A couple of comments / thoughts:
- As we are taking a bit of a "step backwards" by requiring RHEL 7 for the deployment tools, as we have discussed before, we really need to continue working with the BU/engineering teams to find a way to enable Fedora/CentOS runs.
- For the default values - i.e.: in  rhc-ose-ansible/roles/common/defaults/main.yml and rhc-ose-ansible/roles/openstack-create/defaults/main.yml: It would be good to gather all defaults in the same file, or at least the same directory. I know Ansible has it's own structure to follow, but IMHO we should keep these together to avoid confusion and "future bugs".

File: rhc-ose-ansible/roles/openstack-create/defaults/main.yml (hunk 1 above)
Comment (sabre1041) [inline rhc-ose-ansible/roles/openstack-create/defaults/main.yml]: I'll add functionality to perform the lookup of the id and append the branch and will notify when changes have been pushed

**Comment by sabre1041** [comment]
@oybed 
- Agreed. The new docker container and RHEL7 requirement is a temporary solution as discussions continue with the BU
- Still getting my feet wet on "proper" ansible structure. To be honest, some of this is a result of me doing some refactoring but wanted to get the start pushed for others to consume. I am in the midst of working on the storage and continuing to move things around and elevate the areas that will cover each implementation. I still feel that we should separate default variables into their own roles when appropriate (ie openstack vs aws). I'll update the PR with the changes from the comment above relating to flavor lookup to move out as many of the variables as possible out of the openstack role

**Comment by sabre1041** [comment]
@oybed: Updated PR to add support for providing flavor name instead of id. Please give it a whirl when you get time

**Comment by etsauer** [comment]
@sabre1041 i think we need to make the 'openstack-create' role more generic, and maybe even make that a PR of its own. Reason being is that we'll want to reuse the same server creation process for common infrastructure, like cicd, dns, satellite, docker regsitry and other hosts.

**Comment by sabre1041** [comment]
@oybed @etsauer Refactored based on previous comments. OpenStack creation role made to be generic. Created a specific OpenShift provisioning playbook. 

The `common` and `openstack-create` roles can be used for additional playbook use cases (cicd, satellite, registry)

File: rhc-ose-ansible/ose-provision.yml [hunk 2]
```diff
@@ -0,0 +1,37 @@
+---
```
Comment (etsauer) [inline rhc-ose-ansible/ose-provision.yml:1]: @sabre1041 shouldn't this be in a playbook, instead of in the root of the project?

**Comment by sabre1041** [comment]
@etsauer yes it can be in a `playbooks` folder. i was thinking how to include references to roles/plugins etc with playbooks not at the root. i see in [openshift-ansible](https://github.com/openshift/openshift-ansible) they are using symbolic references inside each playbook folder.

**Comment by etsauer** [comment]
@sabre1041 I'm trying to run the playbooks, and continue getting:

```
fatal: [master1-testenv-ycsUMfGY] => SSH Error: Host key verification failed.
    while connecting to 10.3.13.229:22
```

This is because i miss the ssh host key validation. Can't we turn that step off in the ansible.cfg?

**Comment by sabre1041** [comment]
@etsauer are you executing the playbook from the `rhc-ose-ansible` folder? there is an `ansible.cfg` file present that will disable host key checking

**Comment by etsauer** [comment]
@sabre1041 ok, it works. Gonna go ahead and merge.

**Pull request merged**